api/v1beta2/awsmachine_types.go

```diff
@@ -233,6 +233,18 @@ type AWSMachineSpec struct {
 	// If marketType is not specified and spotMarketOptions is provided, the marketType defaults to "Spot".
 	// +optional
 	MarketType MarketType `json:"marketType,omitempty"`
+
+	// HostID specifies the Dedicated Host on which the instance must be started.
+	// +optional
+	HostID *string `json:"hostID,omitempty"`
```

Comment:
Ideally this would set both minimum and maximum length constraints, also, is there a pattern that is valid for this?